test/parallel/test-timers-unenroll-unref-interval.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+const common = require('../common');
+const timers = require('timers');
+
+const keepOpen = setTimeout(() => {
+  common.fail('Test timed out.');
+}, common.platformTimeout(500)); // Keep event loop open.
```

Comment:
> Thanks, I'll fix this so that it more precisely relies on the event loop specifics.

Here's what I'd propose, FWIW:

```js
'use strict';

const common = require('../common');
const timers = require('timers');

setImmediate(() => {}); // Keep event loop open for one tick.

{
  const interval = setInterval(common.mustCall(() => {
    clearTimeout(interval);
    setImmediate(common.mustCall(() => {}));
  }), 1).unref();
}

{
  const interval = setInterval(common.mustCall(() => {
    interval.close();
    setImmediate(common.mustCall(() => {}));
  }), 1).unref();
}

{
  const interval = setInterval(common.mustCall(() => {
    timers.unenroll(interval);
    setImmediate(common.mustCall(() => {}));
  }), 1).unref();
}

{
  const interval = setInterval(common.mustCall(() => {
    interval._idleTimeout = -1;
    setImmediate(common.mustCall(() => {}));
  }), 1).unref();
}

{
  const interval = setInterval(common.mustCall(() => {
    interval._onTimeout = null;
    setImmediate(common.mustCall(() => {}));
  }), 1).unref();
}
```